core/src/main/scala/org/broadinstitute/dsde/rawls/monitor/migration/WorkspaceMigrationActor.scala

```diff
@@ -146,20 +150,70 @@ object WorkspaceMigrationActor {
       }
 
       workspace <- getWorkspace(migration.workspaceId)
-      currentBillingAccountOnGoogleProject <- MigrateAction.liftIO(IO {
+
+      _ <- MigrateAction.raiseWhen(workspace.billingAccountErrorMessage.isDefined) {
+        MigrationException(
+          message = "Workspace migration failed: billing account error exists on workspace",
+          data = Map(
+            ("migrationId" -> migration.id),
+            ("workspaceId" -> workspace.workspaceId),
+            ("billingProjectErrorMessage" -> workspace.billingAccountErrorMessage.get)
+          )
+        )
+      }
+
+      workspaceBillingAccount <- MigrateAction.liftIO(IO {
         workspace.currentBillingAccountOnGoogleProject.getOrElse(
-          throw new RawlsException(s"""No billing account for workspace '${workspace.workspaceId}'""")
+          throw MigrationException(
+            message = "Workspace migration failed: no billing account on workspace",
+            data = Map(
+              ("migrationId" -> migration.id),
+              ("workspaceId" -> workspace.workspaceId)
+            )
+          )
         )
       })
 
+      // Safe to assume that this exists if the workspace exists
+      billingProject <- inTransactionT { _
+        .rawlsBillingProjectQuery
+        .load(RawlsBillingProjectName(workspace.namespace))
+      }
+
+      billingProjectBillingAccount <- MigrateAction.liftIO(IO {
+        billingProject.billingAccount.getOrElse(
+          throw MigrationException(
+            message = "Workspace migration failed: no billing account on billing project",
+            data = Map(
+              ("migrationId" -> migration.id),
+              ("workspaceId" -> workspace.workspaceId),
+              ("billingProject" -> billingProject.projectName)
+            )
+          )
+        )
+      })
+
+      _ <- MigrateAction.raiseWhen(workspaceBillingAccount != billingProjectBillingAccount) {
+        MigrationException(
+          message = "Workspace migration failed: billing account on workspace differs from billing account on billing project",
+          data = Map(
+            ("migrationId" -> migration.id),
+            ("workspaceId" -> workspace.workspaceId),
+            ("workspaceBillingAccount" -> workspaceBillingAccount),
+            ("billingProject" -> billingProject.projectName),
+            ("billingProjectBillingAccount" -> billingProjectBillingAccount)
+          )
+        )
+      }
```

Comment:
Need to add one more check that `RawlsBillingProject.invalidBillingAccount == false` https://github.com/broadinstitute/rawls/blob/93f8e9fb60bf22018bbb2f37e6b5012ea6359fe8/core/src/main/scala/org/broadinstitute/dsde/rawls/model/UserAuth.scala#L56

Reply:
Ah yes that escaped me!